dandi/utils.py

```diff
@@ -709,3 +710,22 @@ def abbrev_prompt(msg: str, *options: str) -> str:
         answer = input(msg).lower()
         if answer in options_map:
             return options_map[answer]
+
+
+def get_mime_type(filename: str, strict: bool = False) -> str:
+    """
+    Like `mimetypes.guess_type()`, except that if the file is compressed, the
+    MIME type for the compression is returned.  Also, the default return value
+    is now ``'application/octet-stream'`` instead of `None`.
+    """
+    mtype, encoding = guess_type(filename, strict)
```

Comment:
It will be awhile until someone propagates the knowledge about the great NWB.   We need to provide ad-hoc handling here for the nwb

```shell
$> python -c 'from mimetypes import guess_type; print(guess_type("sub-RAT123.nwb", False))'
(None, None)

$> file --mime -L sub-RAT123.nwb
sub-RAT123.nwb: application/x-hdf; charset=binary
```

for which we should follow the dandischema in stating it to be `application/x-nwb`:

```
$> git grep encodingFormat                                                                 
dandischema/metadata.py:    if "nwb" in assetmeta["encodingFormat"]:
dandischema/models.py:    encodingFormat: Union[HttpUrl, str] = Field(
dandischema/tests/data/metadata/asset3_01.json:  "encodingFormat": "application/x-nwb",
dandischema/tests/data/metadata/asset3_02.json:  "encodingFormat": "application/x-nwb",
dandischema/tests/data/metadata/asset4_01.json:  "encodingFormat": "application/x-nwb",
dandischema/tests/data/metadata/asset4_02.json:  "encodingFormat": "application/x-nwb",
dandischema/tests/data/metadata/asset_001.json:  "encodingFormat": "application/x-nwb",
dandischema/tests/test_metadata.py:            {"contentSize", "encodingFormat", "id", "identifier", "path", "digest"},
dandischema/tests/test_metadata.py:            {"contentSize", "encodingFormat", "id", "identifier", "path"},
dandischema/tests/test_metadata.py:                "encodingFormat",
dandischema/tests/test_metadata.py:                "encodingFormat",
dandischema/tests/test_metadata.py:                "encodingFormat",
dandischema/tests/test_metadata.py:                "encodingFormat",
dandischema/tests/test_models.py:            contentSize=100, encodingFormat="nwb", digest=digest_model, path="/"
dandischema/tests/test_models.py:            contentSize=100, encodingFormat="nwb", digest=digest_model, path="/"
dandischema/tests/test_models.py:        contentSize=100, encodingFormat="nwb", digest=digest_model, path="/"
dandischema/tests/test_models.py:            contentSize=100, encodingFormat="nwb", digest=digest_model, path="/"
dandischema/tests/test_models.py:            contentSize=100, encodingFormat="nwb", digest=digest_model, path="/"
dandischema/tests/test_models.py:            contentSize=100, encodingFormat="nwb", digest=digest_model, path="/"
```
(@satra -- note that in the tests it is just `encodingFormat="nwb"` and not `encodingFormat="application/x-nwb").

but I wonder if we should be "guess"ing just based on the filename here in general... shouldn't we use proper `libfile/magic` and return `application/x-nwb` ONLY if we do see if being hdf5 and actually have some NWB "marker" (e.g. `h5file.attrs["nwb_version"]`)

Reply:
The function that uses `get_mime_type()` to set `encodingFormat` is only called if `nwb2asset()` fails, which (I think) should only happen if the file in question is not actually an NWB file, so there shouldn't be a need to recognize the `.nwb` extension.